classes/MyTableAdmin.php

```diff
@@ -476,27 +510,31 @@ public function recordSave($messageSuccess = false, $messageError = false)
                     continue;
                 }
                 switch ($field['basictype']) {
-                    case 'integer': case 'rational': case 'choice':
+                    case 'integer':
+                    case 'rational':
+                    case 'choice':
                         if (Tools::among($field['key'], 'PRI', 'UNI') && $original === $value && $value === '') {
                             $value = null;
                         }
                         $sql .= ',' . Tools::escapeDbIdentifier($key) . '='
-                                . (is_null($value) ? 'NULL' : ($field['basictype'] == 'integer' ? (int) $value : (double) $value));
+                            . (is_null($value) ? 'NULL' : ($field['basictype'] == 'integer' ? (int) $value : (double) $value));
                         break;
                     default:
                         $sql .= ',' . Tools::escapeDbIdentifier($key) . '='
-                                . (is_null($value) ? 'NULL' : '"' . $this->escapeSQL($value) . '"');
+                            . (is_null($value) ? 'NULL' : '"' . $this->escapeSQL($value) . '"');
                 }
             }
             $command = 'UPDATE';
+            // todo fix Parameter #1 $types of method GodsDev\MyCMS\MyTableLister::filterKeys() expects array, string given.
             $unique = ($this->filterKeys('PRI') ?: $this->filterKeys('UNI')) ?: array_flip(array_keys($this->fields));
             foreach (array_keys($unique) as $key) {
                 $field = $this->fields[$key];
                 $value = $_POST['fields'][$key];
                 if ($field['key'] == 'PRI' && Tools::among($value, '', null)) {
                     $command = 'INSERT INTO';
                 } else {
-                    $where .= ' AND ' . (is_null($original) ? Tools::escapeDbIdentifier($key) . ' IS NULL' : ($original . '' === '' ? 'IFNULL(' . Tools::escapeDbIdentifier($key) . ', "")' : Tools::escapeDbIdentifier($key)) .' = "' . $this->escapeSQL($_POST['original'][$key]) . '"');
+                    // todo ask CRS2 Variable $original might not be defined.
+                    $where .= ' AND ' . (is_null($original) ? Tools::escapeDbIdentifier($key) . ' IS NULL' : ($original . '' === '' ? 'IFNULL(' . Tools::escapeDbIdentifier($key) . ', "")' : Tools::escapeDbIdentifier($key)) . ' = "' . $this->escapeSQL($_POST['original'][$key]) . '"');
```

Comment:
@crs2 Pokud $this->fields nebylo pole, tak $original nebylo vůbec definováno. Jak to ošetřit?

Reply:
Zkusme změnit druhou větev if v recordSave() z:
if (is_array($this->fields)) {
na:
if (count($this->fields) > 0) {
— $this->fields má být pole s alespoň jedním prvkem

Reply:
Doplnil jsem ještě inicializaci $original = null; a také && isset($command) do podmínky dále. Pokud vše proběhne ok, tak nebudou mít vliv. Pokud by něco selhalo, tak bude kód deterministický.